packages/nextjs/app/components/AnimatedProductDetail.tsx

```diff
@@ -0,0 +1,21 @@
+"use client";
```

Comment:
A few of these added to follow this pattern https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props